### PR TITLE
MAINT: Drop pytest-xdist requirement, minimum pytest version 

### PR DESCRIPTION
**Pull request by effigies**
## Summary

This is an addendum to #2854, following https://github.com/nipy/nipype/pull/2854#issuecomment-454089229. I'm basing off of master instead of @yarikoptic's branch to make sure tests still work. @yarikoptic Feel free to cherry-pick these pieces, if you like, but probably a good idea to wait to make sure that I'm not breaking things horribly. Or if you're happy with this and have nothing further to add, we can just merge this.

@satra @djarecka I believe you had strong feelings on pytest's place in everything, so might be worth a couple minutes of your time.

## List of changes proposed in this PR (pull-request)

* [x] Drop `pytest-xdist` from requirements
* [x] Drop minimum pytest version (was there to support new `pytest-xdist`)
* [x] Adds `pytest-xdist` to Travis and CircleCI builds (via docker, in the latter case) for the sake of speed
* [x] Moves `configparser` requirement into PEP508 format, as right now I think it's being ignored utterly, since we build with Python 3.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,10 +11,6 @@ neurdflib
 click>=6.6.0
 funcsigs
 configparser
-pytest>=3.0
```
[the remark this answers is not in the document]
Comment (effigies) [reply requirements.txt]: I think that will also install the current repo, not just the dependencies, right?

[the remark this answers is not in the document]
Comment (mgxd) [reply requirements.txt]: yep

File: docker/generate_dockerfiles.sh [hunk 2]
```diff
@@ -108,6 +108,8 @@ function generate_main_dockerfile() {
     --miniconda use_env=neuro \
                 pip_opts="-e" \
                 pip_install="/src/nipype[all]" \
```
[the remark this answers is not in the document]
Comment (mgxd) [reply docker/generate_dockerfiles.sh]: whoops, meant L95

File: requirements.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (effigies) [reply requirements.txt]: I don't really care, but that changes what `pip install -r requirements.txt` does. Do we know anybody who installs it that way, and what they desire from that?

@yarikoptic Does Debian use requirements.txt for anything?

File: docker/generate_dockerfiles.sh [hunk 3]
```diff
@@ -92,7 +92,7 @@ function generate_main_dockerfile() {
                 conda_install='python=${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}
                                icu=58.1 libxml2 libxslt matplotlib mkl numpy paramiko
                                pandas psutil scikit-learn scipy traits=4.6.0' \
-                pip_install="grabbit==0.1.2 https://github.com/INCF/pybids/tarball/0.6.5" \
+                pip_install="pytest-xdist" \
```
Comment (effigies) [inline docker/generate_dockerfiles.sh]: I removed the grabbit/pybids stuff, as it's not installed correctly to test `BIDSDataGrabber`, anyway. This should be fixed up in #2737.

File: requirements.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (effigies) [reply requirements.txt]: @mgxd Let's leave changing `requirements.txt` to another PR, where we can seek a consensus on the proper way to manage it.

**Pull request merged**